### PR TITLE
Cache the result of a rule for a LDAP/AD Source to reduce server load.

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Cache the result of a rule for a LDAP/AD Source to reduce server load.
## Impacts

Since the match is cached time based rules will not work properly.
Rules changes can take up to the expire time of the cache to take affect for all users/lookup
## Delete branch after merge

NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Can someone review this?

File: t/data/authentication.conf [hunk 1]
```diff
@@ -47,3 +47,56 @@ description=
...
+
+[LDAP rule Network_Team]
+description=Full Access
+match=any
+action0=set_role=default
+action1=set_access_level=ALL
+action2=set_unreg_date=2022-02-02
+condition0=memberOf,equals,CN=NOC Users,DC=ldap,DC=inverse,DC=ca
+
+[LDAP rule Barb_Iannucci-Web_Designer]
```
Comment (louismunro) [inline t/data/authentication.conf]: You sure this should be in here?

Comment (jrouzierinverse) [inline t/data/authentication.conf]: No

**Comment by louismunro** [comment]
Tested. Ready to be merged.

**Comment by jrouzierinverse** [comment]
Please Merge

**Pull request merged**